homeassistant/components/todoist/calendar.py

```diff
@@ -385,15 +391,15 @@ def __init__(
         self,
         project_data: ProjectData,
         labels: list[Label],
-        api: TodoistAPIAsync,
+        coordinator: TodoistCoordinator,
```

Comment:
This change to use a coordinator generally makes sense. One thing I would like to point out is that the `TodoistProjectData` class is effectively trying to act partially like a data update coordinator. In Google Calendar, which this copied from, I was able to remove this once moving to a coordinator, since it isn't offering much value once you unwind it.

The other thing to consider is that the coordinator is still polling once per minute if I understand correctly. I assume that is more that is needed from a todo list cloud api, but it does mean you get faster updates when things change. The strategy taken by google calendar has always been to have it poll the cloud every 15 minutes, then have the entity itself update every minute so it still advances the clock. (requires a bit of entity adjustment to allow polling at a different interval)

Reply:
That sounds reasonable, I'm not sure what the original intent is for the 1 minute interval was it seems a bit much.  I'll make a note to look into this in the future.  Getting rid of the other class would be great too.

Reply:
The 1 minute interval is likely there just to make sure events start on time. Once you separate the two objects like you have here, now you can tweak the two intervals separately.